server/src/main/kotlin/com/vishnurajeevan/libroabs/Application.kt

```diff
@@ -31,171 +33,244 @@ import kotlin.time.Duration.Companion.days
       """.trimIndent()
-    )
+        )
 
-    runBlocking {
-      val dataDir = File(dataDir).apply {
-        if (!exists()) {
-          mkdirs()
-        }
-      }
-      val tokenFile = File("$dataDir/token.txt")
-      if (!tokenFile.exists()) {
-        lfdLogger("Token file not found, logging in")
-        libroFmApi.fetchLoginData(libroFmUsername, libroFmPassword)
-      }
-
-      libroFmApi.fetchLibrary()
-      processLibrary()
-
-      launch {
-        lfdLogger("Sync Interval: $syncInterval")
-        val syncIntervalTimeUnit = when (syncInterval) {
-          "h" -> 1.hours
-          "d" -> 1.days
-          "w" -> 7.days
-          else -> error("Unhandled sync interval")
-        }
+        runBlocking {
+            val dataDir = File(dataDir).apply {
+                if (!exists()) {
+                    mkdirs()
+                }
+            }
+            val tokenFile = File("$dataDir/token.txt")
+            if (!tokenFile.exists()) {
+                lfdLogger("Token file not found, logging in")
+                libroFmApi.fetchLoginData(libroFmUsername, libroFmPassword)
+            }
 
-        Clock.System.intervalPulse(syncIntervalTimeUnit)
-          .beat { scheduled, occurred ->
-            lfdLogger("Checking library on pulse!")
             libroFmApi.fetchLibrary()
             processLibrary()
-          }
-      }
-
-      serverScope.launch {
-        embeddedServer(
-          factory = Netty,
-          port = port,
-          host = "0.0.0.0",
-          module = {
-            routing {
-              post("/update") {
-                call.respondText("Updating!")
-                libroFmApi.fetchLibrary()
-                processLibrary()
-              }
+
+            launch {
+                lfdLogger("Sync Interval: $syncInterval")
+                val syncIntervalTimeUnit = when (syncInterval) {
+                    "h" -> 1.hours
+                    "d" -> 1.days
+                    "w" -> 7.days
+                    else -> error("Unhandled sync interval")
+                }
+
+                Clock.System.intervalPulse(syncIntervalTimeUnit)
+                    .beat { scheduled, occurred ->
+                        lfdLogger("Checking library on pulse!")
+                        libroFmApi.fetchLibrary()
+                        processLibrary()
+                    }
+            }
+
+            serverScope.launch {
+                embeddedServer(
+                    factory = Netty,
+                    port = port,
+                    host = "0.0.0.0",
+                    module = {
+                        routing {
+                            post("/update") {
+                                call.respondText("Updating!")
+                                libroFmApi.fetchLibrary()
+                                processLibrary()
+                            }
+                            post("/convertToM4b/{isbn}") {
```

Comment:
fun idea. Lets have it respond with "Starting conversion process for <isbn>"

Reply:
Done. Also now lets you know when it wil overwrite files (which is optional)